crates/chia-consensus/src/gen/additions_and_removals.rs

```diff
@@ -0,0 +1,234 @@
+    #[case("block-4671894")]
+    #[case("block-225758")]
+    #[case("block-834752")]
+    #[case("block-834752-compressed")]
+    #[case("block-834760")]
+    #[case("block-834761")]
+    #[case("block-834765")]
+    #[case("block-834766")]
+    #[case("block-834768")]
+    #[case("create-coin-different-amounts")]
+    #[case("create-coin-hint")]
+    #[case("create-coin-hint2")]
+    #[case("duplicate-height-absolute-div")]
+    #[case("just-puzzle-announce")]
+    #[case("many-create-coin")]
+    #[case("many-large-ints-negative")]
+    #[case("max-height")]
+    #[case("multiple-reserve-fee")]
+    #[case("unknown-condition")]
+    fn test_additions_and_removals(#[case] name: &str) {
+        use std::fs::read_to_string;
+
+        let filename = format!("../../generator-tests/{name}.txt");
+        println!("file: {filename}");
+        let test_file = read_to_string(filename).expect("test file not found");
+        let (generator, _expected) = test_file.split_once('\n').expect("invalid test file");
+        let generator = hex::decode(generator).expect("invalid hex encoded generator");
+
+        let mut block_refs = Vec::<Vec<u8>>::new();
+
+        let filename = format!("../../generator-tests/{name}.env");
+        if let Ok(env_hex) = read_to_string(&filename) {
+            println!("block-ref file: {filename}");
+            block_refs.push(hex::decode(env_hex).expect("hex decode env-file"));
+        }
+
+        // we run the block using run_block_generator2() to extract the additions
+        // and removals we *expect* to see
+        // additions_and_removals only work on trusted blocks, so if
+        // run_block_generator2() fails, we can call additions_and_removals() on it.
+        let mut a = make_allocator(0);
+        let conds = run_block_generator2(
+            &mut a,
+            &generator,
+            &block_refs,
+            11_000_000_000,
+            ALLOW_BACKREFS | DONT_VALIDATE_SIGNATURE,
+            &Signature::default(),
+            None,
+            &TEST_CONSTANTS,
+        )
+        .expect("run_block_generator2()");
+
+        let mut expect_additions = HashSet::<(Coin, Option<Bytes>)>::new();
+        let mut expect_removals = HashSet::<Coin>::new();
+
+        for spend in &conds.spends {
+            let removal = Coin {
+                parent_coin_info: a.atom(spend.parent_id).as_ref().try_into().unwrap(),
+                puzzle_hash: a.atom(spend.puzzle_hash).as_ref().try_into().unwrap(),
+                amount: spend.coin_amount,
+            };
+            let coin_id = removal.coin_id();
+            expect_removals.insert(removal);
+            for add in &spend.create_coin {
+                let addition = Coin {
+                    parent_coin_info: coin_id,
+                    puzzle_hash: add.puzzle_hash,
+                    amount: add.amount,
+                };
+                let hint = if add.hint != NodePtr::NIL && a.atom_len(add.hint) <= 32 {
```

Comment:
I think the hint length should be `== 32`

Reply:
it's to maintain consistency with `run_block_generator()`. This isn't a consensus rule, it's just a rule full nodes enforce to make the hint table reasonable.